tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/contact/OpenPaasContactsConsumer.java

```diff
@@ -0,0 +1,138 @@
+                                    RabbitMQConfiguration commonRabbitMQConfiguration,
+                                    EmailAddressContactSearchEngine contactSearchEngine) {
+        this.receiverProvider = receiverProvider;
+        this.sender = sender;
+        this.commonRabbitMQConfiguration = commonRabbitMQConfiguration;
+        this.contactSearchEngine = contactSearchEngine;
+    }
+
+    public void start() {
+        Flux.concat(
+                sender.declareExchange(ExchangeSpecification.exchange(TOPIC)
+                    .durable(DURABLE)),
+                sender.declareQueue(QueueSpecification
+                    .queue(QUEUE_NAME)
+                    .durable(evaluateDurable(DURABLE, commonRabbitMQConfiguration.isQuorumQueuesUsed()))),
+                sender.bind(BindingSpecification.binding()
+                    .exchange(TOPIC)
+                    .queue(QUEUE_NAME)
+                    .routingKey(EMPTY_ROUTING_KEY)))
+            .then()
+            .block();
+
+        consumeContactsDisposable = doConsumeContactMessages();
+    }
+
+    private Disposable doConsumeContactMessages() {
+        return delivery()
+            .subscribeOn(Schedulers.boundedElastic())
+            .subscribe(delivery ->
+                messageConsume(delivery, new String(delivery.getBody(), StandardCharsets.UTF_8)));
+    }
+
+    public Flux<AcknowledgableDelivery> delivery() {
+        return Flux.using(receiverProvider::createReceiver,
+            receiver -> receiver.consumeManualAck(QUEUE_NAME),
+            Receiver::close);
+    }
+
+    private void messageConsume(AcknowledgableDelivery ackDelivery, String messagePayload) {
+        Mono.just(messagePayload)
+            .<ContactAddedRabbitMqMessage>handle((message, sink) -> {
+                try {
+                    sink.next(objectMapper.readValue(message, ContactAddedRabbitMqMessage.class));
+                } catch (JsonProcessingException e) {
+                    sink.error(new RuntimeException(e));
+                }
+            })
+            .handle((msg, sink) -> {
+                handleMessage(msg, sink);
+                ackDelivery.ack();
+            })
+            .onErrorResume(e -> {
+                LOGGER.error("Failed to consume OpenPaaS added contact message", e);
+                return Mono.empty();
+            }).subscribe();
+    }
+
+    private void handleMessage(ContactAddedRabbitMqMessage contactAddedMessage, SynchronousSink<Object> sink) {
+        LOGGER.info("Consumed jCard object message: {}", contactAddedMessage);
+        try {
+            jCardObject jCardObject = contactAddedMessage.vcard();
+            String fullname = jCardObject.fn();
+            MailAddress mailAddress = new MailAddress(jCardObject.email());
+            Username emailAsUsername = Username.fromMailAddress(mailAddress);
+
+            Mono.from(contactSearchEngine.index(AccountId.fromUsername(emailAsUsername),
```

Comment:
Bob creates a contact named "alice" and we add here alice contact into alice address book.

What would be needed here is to resolve the email address of bob (to compute his AccountId) and then add alice contact in it.

For this we need the call to openpaas user API

We can hide such an operation behind an interface to isolate this from the rest of the code here.